app/views/trainees/degrees/_non_uk_degree_form.html.erb

```diff
@@ -2,26 +2,54 @@
 
+  <% if degree_with_invalid_data?(@degree_form.degree) %>
+    <%= render ApplyApplications::ReviewSummary::View.new(form: @degree_form, trainee: @trainee) do |component| %>
+      <% component.header  do %>
+        <%= @apply_invalid_data_view.summary_content %>
+      <% end %>
+
+      <%= @apply_invalid_data_view.summary_items_content %>
+    <% end %>
+  <% end %>
+  
+
   <h1 class="govuk-heading-l">Non-UK degree details</h1>
 
   <%= render FormComponents::CountryAutocomplete::View.new(
-    form_field: f.govuk_collection_select(:country, countries_options, :value,
-                                          :text, label: { text: "In which country is the degree institution based?", size: "s" }),
+    form_field: f.govuk_collection_select(:country, 
+                                          countries_options,
+                                          :value,
+                                          :text,
+                                          form_group: { class: invalid_data_message("country", @degree_form.degree) ? "govuk-form-group govuk-inset-text app-inset-text--narrow-border app-inset-text--important" : "govuk-form-group" },
+                                          label: { text: "In which country is the degree institution based?", size: "s", id: "degrees-country-label" }) do 
+      render InvalidDataText::View.new(degree: @degree_form.degree, form_section: :country)
+    end,
   ) %>
 
   <%= render FormComponents::Autocomplete::View.new(
     f,
     attribute_name: :subject,
     form_field: f.govuk_select(
-      :subject,
+      :subject, 
       options_for_select(subjects_options, @degree_form.degree.subject),
-      label: { text: "Degree subject", size: "s" },
-      hint: { text: "Search for the closest matching subject. You can start typing to narrow down your search." },
-    ),
+      form_group: { 
+        class: invalid_data_message("subject", @degree_form.degree) ? "govuk-form-group govuk-inset-text app-inset-text--narrow-border app-inset-text--important" : "govuk-form-group"
+      },
+      label: { text: "Degree subject", size: "s", id: "degrees-subject-label" },
+      hint: -> do %>
+        <p class="govuk-body govuk-hint"> <%= "Search for the closest matching subject"  %></p> 
+        <%= tag.div(invalid_data_message("subject", @degree_form.degree), class: "app-inset-text__title govuk-!-margin-bottom-3") %>
+      <% end %> 
+    <% ),
   ) %>
 
   <div class="govuk-form-group">
-    <%= f.govuk_radio_buttons_fieldset :non_uk_degree, legend: { text: "Select the UK ENIC comparable degree", size: "s" } do %>
+    <%= f.govuk_radio_buttons_fieldset(:non_uk_degree, legend: { text: "Select the UK ENIC comparable degree", size: "s", id: "degrees-non-uk-label" },  
+        form_group: { class: invalid_data_message("non_uk_degree", @degree_form.degree) ? "govuk-form-group govuk-inset-text app-inset-text--narrow-border app-inset-text--important" : "govuk-form-group" },
+        hint: -> do %>
```

Comment:
This syntax looks a bit funny here or is it just me??

Reply:
updated to use the component instead.